src/Microsoft.AspNetCore.Mvc.Modules/DesignTimeMvcBuilderConfiguration.cs

```diff
@@ -19,19 +21,14 @@ public void ConfigureMvc(IMvcBuilder builder)
             var services = new ServiceCollection();
             services.AddSingleton<IHostingEnvironment>(env);
 
-            var startUp = new Startup(env);
-            startUp.ConfigureServices(services);
-
             serviceProvider = services.BuildServiceProvider();
             var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
 
-            var app = new ApplicationBuilder(serviceProvider);
-            startUp.Configure(app, loggerFactory);
-
-            builder.AddRazorOptions(options =>
+            builder.ConfigureApplicationPartManager(apm =>
             {
-                var extensionLibraryService = services.BuildServiceProvider().GetService<IExtensionLibraryService>();
-                ((List<MetadataReference>)options.AdditionalCompilationReferences).AddRange(extensionLibraryService.MetadataReferences());
+                var extensionLibraryService = serviceProvider.GetRequiredService<IExtensionLibraryService>();
+                apm.FeatureProviders.Add(
+                    new ExtensionMetadataReferenceFeatureProvider(extensionLibraryService.MetadataPaths.ToArray()));
```

Comment:
Anyway here there is a pending PR on `DesignTimeMvcBuilderConfiguration` to be able to resolve shape tag helpers and where we configure services through orchard ... So, we will see ;)